app/v1/controllers/requestParsers/validators/validations/PeriodKeyValidation.scala

```diff
@@ -16,13 +16,13 @@
 
 package v1.controllers.requestParsers.validators.validations
 
-import v1.models.errors.{MtdError, PeriodKeyFormatError}
+import v1.models.errors.{BadRequestError, MtdError, PeriodKeyFormatError}
 
 object PeriodKeyValidation {
 
   private val periodKeyRegex = "^[#0-9A-Z]{4}$"
 
   def validate(periodKey: String): List[MtdError] = {
-    if (periodKey.matches(periodKeyRegex)) NoValidationErrors else List(PeriodKeyFormatError)
+    if (periodKey.matches(periodKeyRegex)) NoValidationErrors else List(PeriodKeyFormatError, BadRequestError)
```

Comment:
Are you doing this to force the error to return a top level invalid request?  There are better ways of doing this.